fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexMaintainer.java

```diff
@@ -552,4 +559,54 @@ public boolean canDeleteWhere(@Nonnull QueryToKeyMatcher matcher, @Nonnull Key.E
         }
         return cursor;
     }
+
+    private static class InstrumentedBunchedMap<K, V> extends BunchedMap<K, V> {
+        @Nonnull
+        private final FDBStoreTimer timer;
+        @Nonnull
+        private final Executor executor;
+
+        public InstrumentedBunchedMap(@Nonnull BunchedMap<K, V> model, @Nonnull FDBStoreTimer timer, @Nonnull Executor executor) {
+            super(model);
+            this.timer = timer;
+            this.executor = executor;
+        }
+
+        @Override
+        protected void instrumentDelete(@Nonnull byte[] key, @Nullable byte[] oldValue) {
+            timer.increment(FDBStoreTimer.Counts.DELETE_INDEX_KEY);
+            timer.increment(FDBStoreTimer.Counts.DELETE_INDEX_KEY_BYTES, key.length);
+            if (oldValue != null) {
+                timer.increment(FDBStoreTimer.Counts.DELETE_INDEX_VALUE_BYTES, oldValue.length);
+            }
+        }
+
+        @Override
+        protected void instrumentWrite(@Nonnull byte[] key, @Nonnull byte[] value, @Nullable byte[] oldValue) {
+            timer.increment(FDBStoreTimer.Counts.SAVE_INDEX_KEY);
+            timer.increment(FDBStoreTimer.Counts.SAVE_INDEX_KEY_BYTES, key.length);
+            timer.increment(FDBStoreTimer.Counts.SAVE_INDEX_VALUE_BYTES, value.length);
+            if (oldValue != null) {
+                // Or should this ignore the value altogether?
+                timer.increment(FDBStoreTimer.Counts.DELETE_INDEX_VALUE_BYTES, oldValue.length);
+            }
+        }
+
+        @Override
+        protected void instrumentRangeRead(@Nonnull CompletableFuture<List<KeyValue>> readFuture) {
+            timer.instrument(FDBStoreTimer.Events.SCAN_INDEX_KEYS, readFuture, executor).whenComplete((list, err) -> {
```

Comment:
`timer.instrument` returns the instrumented future, so I think it probably needs to be returned here.

Reply:
Sure.